sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryClientBuilder.java

```diff
@@ -84,65 +114,90 @@ public ContainerRegistryClientBuilder credential(TokenCredential credential) {
+     *
+     * @param httpPipeline The HTTP pipeline to use for sending service requests and receiving responses.
+     * @return The updated {@link ContainerRegistryClientBuilder} object.
      */
     public ContainerRegistryClientBuilder pipeline(HttpPipeline httpPipeline) {
         this.httpPipeline = httpPipeline;
         return this;
     }
 
     /**
-     * Sets The HTTP client used to send the request.
+     * Sets the HTTP client to use for sending and receiving requests to and from the service.
      *
-     * @param httpClient the httpClient value.
-     * @return the ContainerRegistryBuilder.
+     * @param httpClient The HTTP client to use for requests.
+     * @return The updated {@link ContainerRegistryClientBuilder} object.
      */
     public ContainerRegistryClientBuilder httpClient(HttpClient httpClient) {
         this.httpClient = httpClient;
         return this;
     }
 
     /**
-     * Sets the client options such as application ID and custom headers to set on a request.
+     * Sets the {@link ClientOptions} which enables various options to be set on the client. For example setting an
+     * {@code applicationId} using {@link ClientOptions#setApplicationId(String)} to configure
+     * the {@link UserAgentPolicy} for telemetry/monitoring purposes.
+     *
+     * <p>More About <a href="https://azure.github.io/azure-sdk/general_azurecore.html#telemetry-policy">Azure Core: Telemetry policy</a>
      *
-     * @param clientOptions The {@link ClientOptions}.
-     * @return The updated {@code TableClientBuilder}.
+     * @param clientOptions {@link ClientOptions}.
+     *
+     * @return the updated {@link ContainerRegistryClientBuilder} object
      */
     public ContainerRegistryClientBuilder clientOptions(ClientOptions clientOptions) {
         this.clientOptions = clientOptions;
         return this;
     }
 
     /**
-     * Sets The configuration store that is used during construction of the service client.
+     * Sets the configuration store that is used during construction of the service client.
+     *
+     * The default configuration store is a clone of the {@link Configuration#getGlobalConfiguration() global
+     * configuration store}, use {@link Configuration#NONE} to bypass using configuration settings during construction.
      *
-     * @param configuration the configuration value.
-     * @return the ContainerRegistryBuilder.
+     * @param configuration The configuration store used to
+     * @return The updated {@link ContainerRegistryClientBuilder} object.
      */
     public ContainerRegistryClientBuilder configuration(Configuration configuration) {
         this.configuration = configuration;
         return this;
     }
 
     /**
-     * Sets The logging configuration for HTTP requests and responses.
+     * Sets the logging configuration for HTTP requests and responses.
+     * <p>
+     * If logLevel is not provided, default value of {@link HttpLogDetailLevel#NONE} is set.
```

Comment:
I know this was copied from another client builder, but this should be updated as it is a little confusing. Instead this should call out that if no `HttpLogOptions` are configured that no HTTP request or response logging will happen.

Reply:
No worries, happy to update both.